src/fastaq_functions.py

```diff
@@ -0,0 +1,47 @@
+def check_gc_content(sequence: str, gc_bounds: tuple or float) -> bool:
```

Comment:
Здесь можно подать и int. Если разрешаются 2 типа, то между ними нужно ставить |.
```suggestion
def check_gc_content(sequence: str, gc_bounds: tuple | float | int) -> bool:
```
А еще лучше использовать Union:
```suggestion
def check_gc_content(sequence: str, gc_bounds: Union[tuple,  float, int]) -> bool:
```